src/core/lib/slice/slice.h

```diff
@@ -342,6 +342,22 @@ class GPR_MSVC_EMPTY_BASE_CLASS_WORKAROUND Slice
     return Slice(TakeCSlice());
   }
 
+  // As per TakeOwned, but if the slice is refcounted and there are other refs
+  // then it will copy instead of ref-counting, to ensure the returned slice is
+  // not shared.
+  Slice TakeUniquelyOwned() {
```

Comment:
Do these slices typically tend to be small ? Otherwise is there any danger of too much copying of large slices ?

Reply:
Yeah... we have a default limit of 64kb for *all* of them, so there's an upper bound.
Realistically though, things that get compressed with our workloads are things like call path at 100s of bytes.

Still, there's some variability, so making this an experiment for safety!